compiler/rustc_expand/src/base.rs

```diff
@@ -1394,16 +1394,35 @@ pub(crate) fn pretty_printing_compatibility_hack(nt: &Nonterminal, sess: &ParseS
         if let ast::ItemKind::Enum(enum_def, _) = &item.kind {
             if let [variant] = &*enum_def.variants {
                 if variant.ident.name == sym::Input {
-                    sess.buffer_lint_with_diagnostic(
-                        &PROC_MACRO_BACK_COMPAT,
-                        item.ident.span,
-                        ast::CRATE_NODE_ID,
-                        "using `procedural-masquerade` crate",
-                        BuiltinLintDiagnostics::ProcMacroBackCompat(
-                        "The `procedural-masquerade` crate has been unnecessary since Rust 1.30.0. \
-                        Versions of this crate below 0.1.7 will eventually stop compiling.".to_string())
-                    );
-                    return true;
+                    let filename = sess.source_map().span_to_filename(item.ident.span);
+                    if let FileName::Real(RealFileName::LocalPath(path)) = filename {
+                        if let Some(c) = path
+                            .components()
+                            .flat_map(|c| c.as_os_str().to_str())
+                            .find(|c| c.starts_with("rental"))
+                        {
+                            let mut version = c.trim_start_matches("rental-").split(".");
+                            if version.next() == Some("0")
+                                && version.next() == Some("5")
+                                && version
+                                    .next()
+                                    .and_then(|c| c.parse::<u32>().ok())
+                                    .map_or(false, |v| v < 6)
+                            {
+                                sess.buffer_lint_with_diagnostic(
+                                        &PROC_MACRO_BACK_COMPAT,
+                                        item.ident.span,
+                                        ast::CRATE_NODE_ID,
+                                        "using an old version of `rental`",
+                                        BuiltinLintDiagnostics::ProcMacroBackCompat(
+                                        "older versions of the `rental` crate will stop compiling in future versions of Rust;\
+                                        please update to `rental` v0.5.6".to_string()
```

Comment:
```suggestion
                                        please update to `rental` v0.5.6,, or switch to one of the rental alternatives".to_string()
```
Given how unmaintained `rental` is, I think it is a good idea to expand the suggestion.